util/cpuprofile/cpuprofile.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cpuprofile
+
+import (
+	"bytes"
+	"errors"
+	"runtime/pprof"
+	"sync"
+	"time"
+
+	"github.com/pingcap/tidb/metrics"
+	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/atomic"
+)
+
+var defProfileDuration = time.Second
+
+// globalCPUProfiler is the global CPU profiler.
+// If you want to create a new global cpu profiler, you should stop the old global cpu profiler.
```

Comment:
Maybe you should not ever want anybody to do this, since the rest of your package (for example, your Pprof API compatible layer) does not work well when `globalCPUProfiler` is substituted (for example, the registered consumers just lost) or use a non global `ParallelCPUProfiler`. They simply doesn't work at all. 

So, instead of suggesting someone to stop a global cpu profiler to create a new global one, you should warn people DO NOT EVER try to replace it.

Reply:
Great. I will remove this comment.